### PR TITLE
Save meeting url to metadata of bookings and add it to webhook payload

### LINKED ISSUE #5518: [CAL-498] Include video location in web hooks
### Is your proposal related to a problem?

We want to use the webhook events to save the video location's url (ie google meet / zoom etc)

### Describe the solution you'd like

```
Webhook (Created)
triggerEvent: "BOOKING_CREATED",
payload:{
location: 'integrations:google:meet' //This is where we should include the video link URL for any of the video platforms (Cal, Google Meet, Zoom, etc).
}

Webhook (Rescheduled)
triggerEvent: "BOOKING_RESCHEDULED",
payload:{
location: 'integrations:google:meet' //This is where we should include the video link URL for any of the video platforms (Cal, Google Meet, Zoom, etc).
}

Webhook (Cancelled)
triggerEvent: "BOOKING_CANCELLED",
payload:{
location: 'integrations:google:meet' //This is where we should include the video link URL for any of the video platforms (Cal, Google Meet, Zoom, etc).
}
```

### Describe alternatives you've considered

N/A Feature does not exist currently

### Additional context

This is data point / feature available on other platforms (ie Calendly)

<sub>[CAL-498](https://linear.app/calcom/issue/CAL-498/include-video-location-in-web-hooks)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds new metadata column to booking model to save the meeting URL. Also adds the meeting URL to webhook payload.

Webhook payload for booking created and rescheduled: 

With Cal Video link: 
Screenshot 2022-11-29 at 14 54 32

With Google Meet link: 
Screenshot 2022-11-29 at 14 55 23

Fixes #5518

**Environment**: Staging(main branch)

## How should this be tested?

- Book an event type with a video call app as a location (Cal Video, Google Meet, Zoom, ...)
- Check database if videoCallUrl is saved in metadata 
- Created a webhook and test if it returns the video call data when it is triggered

### CODE REVIEW

**Comment by zomars** [comment]
@CarinaWolli we need to update the test snapshot: 
image

File: packages/prisma/schema.prisma [hunk 1]
```diff
@@ -307,6 +307,7 @@ model Booking {
   smsReminderNumber     String?
   workflowReminders     WorkflowReminder[]
   scheduledJobs         String[]
+  metadata              Json?
```
Comment (zomars) [inline packages/prisma/schema.prisma:310]:
```suggestion
  /// @zod.custom(imports.bookingMetadataSchema)
  metadata              Json?
```

Let's also add this to `packages/prisma/zod-utils.ts`

```ts
export const bookingMetadataSchema = z
  .object({
    videoCallUrl: z.string().optional(),
  })
  .nullable();
```

**Review by zomars — CHANGES_REQUESTED** [review]
This a nice approach @CarinaWolli let's address some concerns before merge first 🙏🏽

**Comment by zomars** [comment]
> > @CarinaWolli we need to update the test snapshot:
> 
> 
> 
> @zomars I have updated them, but they are still failing. I compared the actual and expected test result file and they look identical to me. Do you know how to fix that? 

Run test with `yarn e2e -u` and commit the updated file

File: apps/web/playwright/webhook.e2e.ts [hunk 2]
```diff
@@ -67,12 +67,13 @@ test("add webhook & test that creating an event triggers a webhook call", async
...
   body.payload.videoCallData = dynamic;
   body.payload.appsStatus = dynamic;
+  body.payload.metadata = dynamic;
 
   // if we change the shape of our webhooks, we can simply update this by clicking `u`
   // console.log("BODY", body);
   // Text files shouldn't have platform specific suffixes
   testInfo.snapshotSuffix = "";
-  expect(JSON.stringify(body)).toMatchSnapshot(`webhookResponse.txt`);
+  expect(JSON.stringify(body) + "\n").toMatchSnapshot(`webhookResponse.txt`);
```
Comment (zomars) [inline apps/web/playwright/webhook.e2e.ts:76]: Let's revert this change.

File: apps/web/playwright/webhook.e2e.ts [hunk 3]
```diff
@@ -67,12 +67,13 @@ test("add webhook & test that creating an event triggers a webhook call", async
   body.payload.eventTypeId = dynamic;
   body.payload.videoCallData = dynamic;
   body.payload.appsStatus = dynamic;
+  body.payload.metadata = dynamic;
```
Comment (zomars) [inline apps/web/playwright/webhook.e2e.ts:70]: Do we want to redact metadata?

**Review by zomars — CHANGES_REQUESTED** [review]
Left some comments

File: apps/web/playwright/webhook.e2e.ts (hunk 3 above)
Comment (CarinaWolli) [reply apps/web/playwright/webhook.e2e.ts:70]: I am not exactly sure what you mean with that?

Comment (zomars) [reply apps/web/playwright/webhook.e2e.ts:70]: Redacted data applies for things we don't care what value have (like meeting urls) If we expected certain metadata to be present we may want to know that in the test.

Comment (CarinaWolli) [reply apps/web/playwright/webhook.e2e.ts]: Updated snapshot to: "metadata":{"videoCallUrl":"[redacted/dynamic]"}

**Review by zomars — APPROVED** [review]
Solved conflicts and ready to go

LGTM
